hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -0,0 +1,220 @@
+    return st.decimals(min_value=-bound, max_value=bound, places=field.decimal_places)
+
+
+@register_for(dm.CharField)
+@register_for(dm.TextField)
+def _for_text(field):
+    # We can infer a vastly more precise strategy by considering the
+    # validators as well as the field type.  This is a minimal proof of
+    # concept, but we intend to leverage the idea much more heavily soon.
+    # See https://github.com/HypothesisWorks/hypothesis-python/issues/1116
+    regexes = [
+        re.compile(v.regex, v.flags) if isinstance(v.regex, str) else v.regex
+        for v in field.validators
+        if isinstance(v, django.core.validators.RegexValidator) and not v.inverse_match
+    ]
+    if regexes:
+        # This strategy generates according to one of the regexes, and
+        # filters using the others.  It can therefore learn to generate
+        # from the most restrictive and filter with permissive patterns.
+        # Not maximally efficient, but it makes pathological cases rarer.
+        # If you want a challenge: extend https://qntm.org/greenery to
+        # compute intersections of the full Python regex language.
+        return st.one_of(*[st.from_regex(r) for r in regexes])
+    # If there are no (usable) regexes, we use a standard text strategy.
+    return st.text(
+        alphabet=st.characters(
+            blacklist_characters=u"\x00", blacklist_categories=("Cs",)
+        ),
+        min_size=(0 if field.blank else 1),
+        max_size=field.max_length,
+    )
+
+
+def register_field_strategy(field_type, strategy):
+    # type: (Type[dm.Field], st.SearchStrategy) -> None
+    """Add an entry to the global field-to-strategy lookup used by from_field.
+
+    ``field_type`` must be a subtype of django.db.models.Field, which must not
+    already be registered.  ``strategy`` must be a SearchStrategy.
+    """
+    if not issubclass(field_type, dm.Field):
+        raise InvalidArgument(
+            "field_type=%r must be a subtype of Field" % (field_type,)
+        )
+    check_type(st.SearchStrategy, strategy, "strategy")
+    if field_type in _global_field_lookup:
+        raise InvalidArgument(
+            "field_type=%r already has a registered strategy (%r)"
+            % (field_type, _global_field_lookup[field_type])
+        )
+    if issubclass(field_type, dm.AutoField):
+        raise InvalidArgument("Cannot register a strategy for an AutoField")
+    _global_field_lookup[field_type] = strategy
+
+
+def from_field(field):
+    # type: (Type[dm.Field]) -> st.SearchStrategy[dm.Field]
+    """Return a strategy for values that fit the given field.
+
+    This is pretty similar to the core `from_type` function, with two subtle
+    but important differences.  First, `from_field` takes a Field *instance*,
+    rather than a Field *subtype*, so that it has access to instance attributes
+    such as string length and validators.  Second, while examples drawn from
```

Comment:
I'm not sure the second part needs to be said - this is pretty universally true of *all* Hypothesis strategies.